qiskit/circuit/library/generalized_gates/qft.py

```diff
@@ -0,0 +1,53 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""QFTGate: gate class for natively reasoning about Quantum Fourier Transforms."""
+
+from __future__ import annotations
+import numpy as np
+from qiskit.circuit.quantumcircuit import Gate
+
+
+class QFTGate(Gate):
```

Comment:
This class appears in the Section "Generalized Gates", unlike QFT which appears in the section "Basis Change Circuits". 
Is it deliberate?

Reply:
Yes! It's a gate but not a "basic gate", so it's a "generalized gate". Are you proposing to restructure the library of generalized gates, grouping the generalized gates by "their purpose"?

Reply:
I'm actually not sure why QFT appears separately than the other generalized gates

Reply:
I think that maybe all the (multi/uniformly) controlled gates deserve a special category (but that's not relevant to this PR)

Reply:
Sure. I have moved the `QFTGate` to `basis_change`, the same file that contains the `QFT` circuit.